src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -132,7 +186,13 @@ public ObservableList<Person> getFilteredPersonList() {
     @Override
     public void updateFilteredPersonList(Predicate<Person> predicate) {
         requireNonNull(predicate);
-        filteredPersons.setPredicate(predicate);
+        if (isViewingArchivedList) {
+            // Apply the predicate only to archived persons if viewing archived list
+            filteredPersons.setPredicate(person -> predicate.test(person) && person.isArchived());
```

Comment:
```suggestion
            filteredPersons.setPredicate(person -> person.isArchived() && predicate.test(person));
```

Reply:
Minor detail but it might be better to check isArchived() first due to short circuit evaluation. Same for line 194